File: Leetcode/3396. Minimum Number of Operations to Make Elements in Array Distinct.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        int n=nums.size(),i;
        unordered_set<int> freq;
        bool duplicate=false;
        for( i=n-1;i>=0;i--){
            if(freq.count(nums[i])==0){
                freq.insert(nums[i]);
            }else{
                duplicate=true;
                break;
            }
        }
        if(i==0 && duplicate==false){
            return 0;
        }
        return (i+3)/3;
        
    }
};

File: Leetcode/3396. Minimum Number of Operations to Make Elements in Array Distinct.py
class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        n=len(nums)
        freq=set()
        duplicate=False
        for i in range(n-1,-1,-1):
            if nums[i] not in freq:
                freq.add(nums[i])
            else:
                duplicate=True
                break
        if (i==0) and (duplicate==False):
            return 0
        return (i+3)//3

        
